clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -501,18 +461,51 @@ int skipCmakeComments(const std::string &Input, int Index) {
   return Index;
 }
 
-void cmakeSyntaxProcessed(
-    std::string &Input,
-    const std::map<std::string, std::string> &VariablesMap) {
+void processCmakeMinimumRequired(std::string &Input, size_t &Size,
+                                 size_t &Index) {
+  std::string VarName;
+  std::string Value;
+  size_t Begin, End;
+
+  // Get the begin of firt argument of cmake_minimum_required
+  Index = skipWhileSpaces(Input, Index);
+  Begin = Index;
 
-  int Size = Input.size();
-  int Index = 0;
+  // Get the end of the first argument of cmake_minimum_required
+  Index = gotoEndOfCmakeWord(Input, Index, ')');
+  End = Index;
+
+  // Get the name of the first argument
+  VarName = Input.substr(Begin, End - Begin);
+
+  // Get the begin of the second argument
+  Index = skipWhileSpaces(Input, End + 1);
+  Begin = Index;
+
+  // Get the end of the second argument
+  Index = gotoEndOfCmakeWord(Input, Begin + 1, ')');
+  End = Index;
+
+  // Get the name of the second argument
+  Value = Input.substr(Begin, End - Begin);
+
+  std::string ReplStr = processArgOfCmakeVersionRequired(Value, CmakeVarMap);
+
+  Input.replace(Begin, End - Begin, ReplStr);
+  Size = Input.size();            // Update string size
+  Index = Begin + ReplStr.size(); // Update index
+}
+
+void applyImplicitMigrationRules(std::string &Input) {
```

Comment:
Please add comment on: implict migration Rules is used in what kind of scenarsio.

Reply:
ImplicitMigrationRules is used when the migration logic is difficult to be described with yaml based rule syntax.
Currently, only migration of cmake_minimum_required() is applied with ImplicitMigrationRule. Once ImplicitMigrationRule for some cmake syntax is implemented, the yaml based rules for the same cmake syntax will be skipped and a msg is emitted to notify the user.